pkg/pillar/zedUpload/datastore_aws.go

```diff
@@ -193,6 +196,35 @@ func (ep *AwsTransportMethod) processS3Download(req *DronaRequest) (error, int)
 	return err, csize
 }
 
+func (ep *AwsTransportMethod) processS3DownloadByChunks(req *DronaRequest) error {
+	pwd := strings.TrimSuffix(ep.apiKey, "\n")
+	dataChan := make(utils.ChunkDataChan)
+	defer close(dataChan)
+	go func(req *DronaRequest, dataNotif utils.ChunkDataChan) {
+		for {
+			select {
+			case stats, ok := <-dataNotif:
+				if !ok {
+					return
+				}
+				ep.ctx.postChunk(req, stats.Size, stats.Chunk)
+			}
+		}
+	}(req, dataChan)
+
```

Comment:
Same comments as for the Azure function below. 
BTW: I typically use identical names for functions which are variants; makes it easier to understand and see the differences even though one invokes S3 and the other Azure APIs,

Reply:
functions renamed